### DISCUSSION TITLE
`ggsave` is confusing

### DISCUSSION THREAD

**Discussion by marcpabst** [JetBrains/lets-plot#1163] [General]
Is there a good way to make sure that 
- (a) my plot actually ends up being the apsect ratio I set  using the `w` and `h` paramters of `ggsave`, and
- (b) my 15pt font size is actually 15pt in an exported PDF? 

So far, my plot adheres to dome magic, pre-set aspect ratio and I end up with lots of white space in my PDF. I can use `ggsize` to change that, but that screws up the scaing.

**Comment by alshan** [comment]
Yes, you are supposed to control the plot's aspect ratio by setting its absolute size in pixels using `ggsize()`. What's happening with the scaling?

**Comment by marcpabst** [comment]
It's a bit unclear how pixels in `ggsize` relate to physical units in `gsave`. I would have assumed that you can control that via the `dpi` paramter, but that seems to do nothing. I think per default, 72px = 1inch?

**Comment by alshan** [comment]
Maybe 1 pixel = 1/96th as was mentioned in this discussion: https://github.com/JetBrains/lets-plot/issues/839
But we actually don't do this conversion directly. We only create the plot SVG (where the size is in pixels) and then pass it to  Cairosvg (along with the desired dimensions in inches) to generate PNG or PDF. 

UPD: we also pass provided DPI to CairoSVG so I'm pretty sure the px/inch ratio is controlled by the DPI.

**Comment by alshan** [comment]
We've updated the [`ggsave()`](https://lets-plot.org/python/pages/api/lets_plot.ggsave.html) documentation (see `Notes`) and related cookbook [here](https://nbviewer.org/github/JetBrains/lets-plot-docs/blob/master/source/examples/cookbook/export.ipynb) (see "Parameters w, h, unit, and dpi" section).
I hope this address this issue well.
